admin/tabs/general/roles/details/members/RoleMembersModel.ts

```diff
@@ -101,13 +98,15 @@ export class RoleMembersModel extends HoistModel {
             return;
         }
 
-        if (this.props.showEffective) {
+        if (!this.props.showInherited) {
             this.gridModel.loadData([
                 // 1 - Users
                 ...role.effectiveUsers.map(it => ({
                     name: it.name,
                     sources: this.sortThisRoleFirst(it.sources),
-                    type: RoleMembersModel.types.USER
+                    type: RoleMembersModel.types.USER,
+                    dateCreated: role.members.find(member => member.name === it.name)?.dateCreated,
```

Comment:
As discussed, would make a collection of members keyed by name so you don't have to search multiple times per cycle

Reply:
Also should be matching by name and type. It's unlikely 2 members of different types would have the same name, but I don't believe we forbid it.